lib/taskana-core/src/test/java/pro/taskana/ArchitectureTest.java

```diff
@@ -84,20 +83,92 @@ void noClassShouldAccessStandardStreams() {
     NO_CLASSES_SHOULD_ACCESS_STANDARD_STREAMS.check(importedClasses);
   }
 
-  @Test
+  /**
+   * Solution would be nice, but is not achievable without big changes
+   * https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks
+   */
   @Disabled
+  @Test
   void freeOfCycles() {
-    ArchRule myRule = slices().matching("pro.taskana.(*)..")
-                          .should().beFreeOfCycles();
+    ArchRule myRule = slices().matching("pro.taskana.(*)..").should().beFreeOfCycles();
     myRule.check(importedClasses);
   }
 
+  /**
+   * Test for cycles with subpackages
+   * https://www.archunit.org/userguide/html/000_Index.html#_cycle_checks
+   */
+  @TestFactory
+  Collection<DynamicTest> freeOfCycles_subpackages() {
+
+    Stream<String> packagesToTest =
+        Stream.of(
+            "pro.taskana.common.internal.(*)..",
+            "pro.taskana.common.api.(*)..",
+            "pro.taskana.common.internal.(*)..",
+            "pro.taskana.classification.api.(*)..",
+            "pro.taskana.classification.internal.(*)..",
+            "pro.taskana.history.api.(*)..",
+            "pro.taskana.history.internal.(*)..",
+            // to be fixed soon
+            // "pro.taskana.report.api.(*)..",
+            "pro.taskana.report.internal.(*)..",
+            "pro.taskana.task.api.(*)..",
+            "pro.taskana.task.internal.(*)..",
+            "pro.taskana.workbasket.api.(*)..",
+            "pro.taskana.workbasket.internal.(*)..");
+    return packagesToTest
+        .map(
+            p ->
+                dynamicTest(
+                    p.replaceAll(Pattern.quote("pro.taskana."), "") + " is free of cycles",
+                    () -> slices().matching(p).should().beFreeOfCycles().check(importedClasses)))
+        .collect(Collectors.toList());
+  }
+
+  @Disabled("TBD")
   @Test
+  void commonClassesShouldNotDependOnOtherDomainClasses() {
+    ArchRule myRule =
+        noClasses()
+            .that()
+            .resideInAPackage("..common..")
+            .should()
+            .dependOnClassesThat()
+            .resideInAnyPackage(
+                "..classification..", "..history..", "..report..", "..task..", "..workbasket..");
+    myRule.check(importedClasses);
+  }
+
   @Disabled
-  void freeOfCyclicDependencies() {
-    ArchRule myRule = slices().matching("pro.taskana.(*)..").should()
-                          .notDependOnEachOther();
+  @Test
+  void noClassesShouldDependOnReports() {
```

Comment:
Das Problem scheint zu sein, dass der DaysToWorkingDaysConverter, der aktuell im report package liegt, von TaskServiceImpl benutzt wird. Vielleicht sollten wir diesen in das common package refactoren.
Dann könnten wir diesen Test enablen...